letsencrypt-nginx/letsencrypt_nginx/tests/util.py

```diff
@@ -85,3 +85,18 @@ def traverse(tree):
                     yield [key, values]
 
     return list(traverse(tree))
+
+def contains_at_depth(haystack, needle, n):
+    """
+    Return true if the needle is present in one of the sub-iterables in haystack
+    at depth n. Haystack must be an iterable.
+    """
+    if not hasattr(haystack, '__iter__'):
```

Comment:
I tried this and wound up going back: I specifically depend on strings not having the __iter__ attribute. Added a comment.